### PR TITLE
fix: fix StopIteratorException

### PR DESCRIPTION
**Pull request by wochinge**
### Related Issues

- fixes https://github.com/deepset-ai/deepset-cloud-sdk/issues/36

### Proposed Changes?
- introduce util to cleanly iterate over async generator
- adjust project to target lowest compatible version (otherwise we might not catch some issues with older versions)

### Checklist

- [ ] I have updated the referenced issue with new insights and changes
- [ ] If this is a code change, I have added unit tests
- [ ] I've used the [conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title
- [ ] I updated the docstrings
- [ ] If this is a code change, I added meaningful logs and prepared Datadog visualizations and alerts

### CODE REVIEW

File: deepset_cloud_sdk/workflows/sync_client/utils.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+def iter_over_async(ait, loop):
```
Comment (wochinge) [inline deepset_cloud_sdk/workflows/sync_client/utils.py:1]: todo: typing +tests

File: deepset_cloud_sdk/workflows/sync_client/utils.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    # Stolen from https://stackoverflow.com/questions/63587660/yielding-asyncio-generator-data-back-from-event-loop-possible/63595496#63595496
```
[the remark this answers is not in the document]
Comment (rjanjua) [reply deepset_cloud_sdk/workflows/sync_client/utils.py:2]: [here are some lovely alternatives](https://www.thesaurus.com/browse/stolen) 😜

File: pyproject.toml [hunk 3]
```diff
@@ -78,11 +78,12 @@ dependencies = [
 
 
 [tool.hatch.envs.code-quality]
+python = "3.8"
 template='default'
 detached = false
 # Please keep these aligned with the versions defined in .pre-commit-config.yaml
 dependencies = [
-  "pylint==2.15.2",
+  "pylint==2.17.4",
```
Comment (wochinge) [inline pyproject.toml:86]: needed updating to fix false positive

**Review by rjanjua — APPROVED** [review]
Looks good 👍. Thanks for updating the version of python we run our CI with too 🙌

**Pull request merged**